Kuznetsov pavel

Technic47 opened the pull request:
Всё скопировано с другой папки*, так как сначала сделал всю работу, а уже потом сделал fork( Комитов не видно.
Реализовал генератор поколений на основе классов Human и Parent. Всё хранится в базе. Не успел реализовать систему записи и считывания. Там разбираться ещё на пару дней. Поигрался с static методами и классами целиком. Private, public, static освоил. Cast классов не понимаю. Как-то сделал, как-то работает.

В папке Practice from Lection1 добавлен проект по лекции1 с магами и т.п.
В нём попробовал реализовать Healer и поработать с методами для использования расходников.

Technic47 committed:
Всё скопировано с другого ресурса, так как сначала сделал вю работу, а уже потом сделал fork(
В папке Practice from Lection1 добавлен проект по лекции1 с магами и т.п.
В нём попробовал реализовать Healer и поработать с методами для использования расходников.

Technic47 committed:
Всё скопировано с другого ресурса, так как сначала сделал вю работу, а уже потом сделал fork(
В папке Practice from Lection1 добавлен проект по лекции1 с магами и т.п.
В нём попробовал реализовать Healer и поработать с методами для использования расходников.

Liberate520 commented on Practice from Lection1/src/BaseHero.java:
```diff
@@ -0,0 +1,95 @@
+        return consumables > 0;
+    }
+
+    private void healed(int Hp) {
```
Неверно) в базовом классе не должно быть такого метода, так как этот метод не присущ всем дочерним классам)

Liberate520 replied:
Хотя могла бы быть какая то ability, уникальная способность. Сделать метод абстрактным с идеей того, что у каждого класса уникальная способность может быть своя, но это только если мы решили, что обязательно у всех героев должна быть уникальная способность

Liberate520 commented on src/analizator/Treeanalizer.java:
```diff
@@ -0,0 +1,114 @@
+                    System.out.println("There are no siblings in this family.");
+                } else {
+                    for (Human child : children) {
+                        if (!Objects.equals(child.getName(), person.getName())) { // проверка на схожесть имён
```
Можно было переопределить метод equals

Liberate520 commented on src/dataBase/DataBase.java:
```diff
@@ -0,0 +1,151 @@
+import java.util.ArrayList;
+
+public class DataBase {
+    ArrayList<Human> db;
```
закрыть поля для прямого взаимодействия, сделать приватными

Liberate520 commented on src/dataBase/DataBase.java:
```diff
@@ -0,0 +1,151 @@
+    /*
+    возвращает список записей имеющих определённый статус гендера
+     */
+    public ArrayList<Human> getListOf(state state) {
```
Не вижу файла State, видимо это enum, но его почему то нет. В любом случае название Enum пишется с большой буквы, то есть State state

Technic47 committed:
Внесены правки

Technic47 committed:
refactor

Technic47 committed:
Huge refactor.
Interface implemented.

Technic47 committed:
Double siblings printing problem.

Technic47 committed:
Всё работает, проблем не обнаружено.

Technic47 committed:
Рефактор интерфейса.
Разобрался с default методами. Переписал некоторые абстрактные.

Technic47 committed:
Удалил пример с магами из лекции 1.
Нет смысла с ними возиться.

Technic47 commented:
Переписал много чего. 
- Вынес enum в отдельные классы.
- Переработал работу с FamilyTree. Теперь все записи там хранятся по паре <RelationType, ArrayList<Parent>>. Не совсем понимаю, может ли HashSet хранить несколько записей по одному ключу. Пишут, что нет, но в базе, похоже, что хранятся. Сделал лист на случай нескольких детей одного пола.
- Сделал интерфейс, попробовал как default методы, так и  абстратные.

Technic47 committed:
Добавлены классы в анализатор.
Реализованы интерфейсе Iterator\Iterable и Comparator.
Семьи сортируются по количество потомков и перебораются через foreach.

Technic47 commented:
Работа по 3му семинару проведена в пакете analizator.
Создал отдельный класс Pair в котором хранится семейная пара из 2х объектов.
В FamiliesAnalizer хранится список со всеми Pair и там же имплементирован итератор.
В методе sortByDescendants использован компаратор по полю количества потомков у Pair.

Очень понравилась сома идея создания своих компараторав и итераторов. Ваши разъяснения прояснили что к чему, спасибо!

Technic47 committed:
Small refactor

Technic47 committed:
Сделан контроллер комманд.
Сделан блок IO, отвечающий за запись и чтение абстрактного объекта.

Technic47 commented:
Обновления для дз по 4-ому семинару.

Technic47 committed:
Works fine

Technic47 committed:
Works fine

Technic47 committed:
Refactor

Technic47 committed:
Final refactor without pattern.

Technic47 committed:
MVP pattern try.

Technic47 committed:
Refactor paths in Service.

Technic47 committed:
Схема переделана

Technic47 commented:
Передалал под MVP. Вроде, соответствует.
Dispatcher принимает команды из консили и отправляет через Presenter всё в Service. А тот уже занимается важными делами по обходу модулей и отправки обратно статистики.
Статистика через Presenter  попадает в Dispatcher и выводится на экран.

Liberate520 commented on src/presenter/Presenter.java:
```diff
@@ -0,0 +1,27 @@
+        this.service = service;
+    }
+
+    private Service service;
```
Все поля принято писать в начале тела класса

Liberate520 commented on src/presenter/Presenter.java:
```diff
@@ -0,0 +1,27 @@
+
+    private Service service;
+
+    public Presenter(Dispatcher dispatcher){
```
Должен принимать обхект абстракции, т.е. View

Liberate520 commented on src/service/dataBase/ParentIterator.java:
```diff
@@ -0,0 +1,25 @@
+import java.util.ArrayList;
+import java.util.Iterator;
+
+public class ParentIterator implements Iterator<Person> {
```
видимо класс должен был называться DataBaseIterator

Technic47 committed:
Переопределены Equals и hashcode

Technic47 committed:
Откат

Technic47 committed:
Refactor

Technic47 committed:
Refactor.
SOLID try.
Timings added.

Technic47 committed:
Refactor.
SOLID try.
Analyzers rebuilt.

Technic47 committed:
Refactor statistics

Technic47 committed:
Refactor person Analizer. Extended added.

Technic47 committed:
Refactor.
Comments added to classes.

Technic47 committed:
db.dat cleared.

Technic47 committed:
refactor

Technic47 committed:
refactor

Technic47 committed:
refactor DBanalizer in service

Technic47 committed:
refactor IOwork rebuilt

Technic47 committed:
refactor IO comments
